federatedscope/core/fed_runner.py

```diff
@@ -156,7 +156,13 @@ def _setup_server(self):
         """
         self.server_id = 0
         if self.mode == 'standalone':
-            if self.server_id in self.data:
+            if self.cfg.federate.merge_test:
+                from federatedscope.core.auxiliaries.data_builder import merge_test_data
+                num_of_sample, server_data = merge_test_data(all_data=self.data)
+                model = get_model(self.cfg.model,
+                              server_data,
```

Comment:
It will report error when initialize a trainer within the server, since the `get_cnn` function requires some data from 'train' (federatedscope/cv/model/model_builder.py) to initialize the CNN models but the server only has 'test' data.

Reply:
Also, the number of training data (`ctx.num_train_data=0`) in the server will cause some error for dp algorithm [`NBAFL` ](https://github.com/alibaba/FederatedScope/blob/master/federatedscope/core/trainers/trainer_nbafl.py#L65).